### PR TITLE
[MRG] init parameter validation tests

### PR DESCRIPTION
**Pull request by NicolasHug**
Still more to do (in other PRs I think), but here are basic checks

### CODE REVIEW

File: pygbm/gradient_boosting.py [hunk 1]
```diff
@@ -68,9 +91,11 @@ def fit(self, X, y):
             duration = toc - tic
             troughput = X.nbytes / duration
             print(f"{duration:.3f} s ({troughput / 1e6:.3f} MB/s)")
+
         if self.validation_split is not None:
+            # TODO: stratify for classification
```
Comment (ogrisel) [inline pygbm/gradient_boosting.py]: We could check if `self.loss_` has a `predict_proba` method.

File: pygbm/gradient_boosting.py [hunk 2]
```diff
@@ -33,9 +34,37 @@ def __init__(self, loss, learning_rate, max_iter, max_leaf_nodes,
         self.tol = tol
         self.verbose = verbose
         self.random_state = random_state
-        self.loss = loss
+
+    def _validate_parameters(self):
+
+        if self.loss not in _LOSSES:
+            raise ValueError("Invalid loss {}. Accepted losses are {}.".format(
+                self.loss, ', '.join(self._SUPPORTED_LOSS)))
```
[the remark this answers is not in the document]
Comment (NicolasHug) [reply pygbm/gradient_boosting.py]: yes this is done in #49 , I'll update the test there once this PR is merged to master?

**Comment by NicolasHug** [comment]
Thanks for the review, I addressed the comments.

Currently we check the `__init__` parameters "as deep as possible" in the code, that is e.g. for the `max_leaf_nodes` parameter, we test it in `TreeGrower` because it's only used there, even though it's initially passed to `BaseGradientBoostingMachine`.

But in the tests, the `assert_raises_...` test for `max_leaf_nodes` is done in `test_gradient_boosting.py` and not in `test_grower` because it is first exposed to the user in `BaseGradientBoostingMachine`, not in `TreeGrower`.

That was my logic but LMK if you'd rather have it another way. No strong opinion on my side.

Note that we cannot make input validation in `SplittingContext` as it's a jitclass so it's done in `TreeGrower` instead.

File: pygbm/gradient_boosting.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ogrisel) [reply pygbm/gradient_boosting.py]: Ok

**Comment by ogrisel** [comment]
Merged.

**Pull request merged**